apps/s_server.c

```diff
@@ -470,7 +470,7 @@ static int ssl_servername_cb(SSL *s, int *ad, void *arg)
         BIO_printf(p->biodebug, "Hostname in TLS extension: \"");
         while ((uc = *cp++) != 0)
             BIO_printf(p->biodebug,
-                       isascii(uc) && isprint(uc) ? "%c" : "\\x%02x", uc);
+                       (((uc) & ~127) == 0) && isprint(uc) ? "%c" : "\\x%02x", uc);
```

Comment:
I would not mind removing isascii completely in this case, since uc is unsigned char.
When I brought up the issue with isxxx I was making a point about an implementation where
isprint is a macro which is assuming the argument is in the range -1 .. 255, thus -1 being EOF.
and that breaks when using signed char as index.